src/Tooltip/__tests__/Tooltip.spec.js

```diff
@@ -0,0 +1,169 @@
+  return describe(`on ${eventType}`, () => {
+    let button, tooltip, themeProvider, trigger;
+
+    beforeEach(() => {
+      [themeProvider, tooltip] = mountTooltip({ onOpen: jest.fn() });
+      trigger = tooltip.find(PopoverTrigger);
+      button = trigger.find('button');
+
+      jest.useFakeTimers();
+      button.simulate(eventType);
+      jest.runAllTimers();
+    });
+
+    it('renders content', () => {
+      const content = themeProvider.update().find(PopoverContent);
+      expect(content.exists()).toEqual(true);
+    });
+
+    it('calls onOpen', () => {
+      expect(tooltip.props().onOpen).toHaveBeenCalled();
+    });
+
+    it('without setting focus to the trigger', () => {
+      assertTriggerHasFocus(trigger, false);
+    });
+  });
+}
+
+function testClosingOnEvent(eventType) {
+  return describe(`on ${eventType}`, () => {
+    let button, tooltip, themeProvider, trigger;
+
+    beforeEach(() => {
+      [themeProvider, tooltip] = mountTooltip({
+        onClose: jest.fn(),
+        defaultIsOpen: true
+      });
+      trigger = tooltip.find(PopoverTrigger);
+      button = trigger.find('button');
+      button.simulate(eventType);
+    });
+
+    it('closes the content', () => {
+      const content = themeProvider.find(PopoverContent);
+      expect(content.exists()).toEqual(false);
+    });
+
+    it('calls onClose', () => {
+      expect(tooltip.props().onClose).toHaveBeenCalled();
+    });
+  });
+}
+
+describe('Tooltip', () => {
+  describe('renders', () => {
+    it('root', () => {
+      const tooltip = shallowTooltip();
+
+      expect(tooltip.exists()).toEqual(true);
+    });
+
+    it('disregards hasArrow prop', () => {
+      const [, tooltip] = mountTooltip({
+        hasArrow: false,
+        defaultIsOpen: true
+      });
+      const arrow = tooltip.find(PopoverArrow);
+
+      expect(arrow.exists()).toEqual(true);
+    });
+  });
+
+  describe('opens', () => {
+    ['mouseenter', 'focus'].forEach(testOpeningOnEvent);
+  });
+
+  describe('closes', () => {
+    ['mouseleave', 'blur'].forEach(testClosingOnEvent);
+
+    it('when pressing escape', () => {
+      const [themeProvider, tooltip] = mountTooltip({
+        onClose: jest.fn(),
+        defaultIsOpen: true
+      });
+      const trigger = tooltip.find(PopoverTrigger);
+
+      const event = new KeyboardEvent('keydown', {
+        key: 'Escape',
+        view: window,
+        bubbles: true,
+        cancelable: true
+      });
+
+      document.body && document.body.dispatchEvent(event);
+      themeProvider.update();
+
+      const content = themeProvider.find(PopoverContent);
+
+      expect(content.exists()).toEqual(false);
+      assertTriggerHasFocus(trigger);
+    });
+  });
+
+  describe('disabled', () => {
+    it('does not open on click', () => {
+      const [themeProvider, tooltip] = mountTooltip({ disabled: true });
+      const trigger = tooltip.find(PopoverTrigger);
+      trigger.find('button').simulate('mouseenter');
```

Comment:
This test is labeled "on click", but the tested event is `mouseenter`.

Reply:
Updated.